### PR TITLE
[#34] Add localization feature to manage translations and reuse labels

### PR DESCRIPTION
**Pull request by malparty**
## What happened 

Configure i18next to manage locales

## Insight

- mocks enable to test using the translation slug instead of the real translation. You can update/fix any translation without breaking your tests.
- Autocompletion is now also available in the `t('translation_key')` calls and `useTranslate('NAMESPACE')` hook

### Locales are working:

|The landing page uses locales | Translation file | Code |
|---|---|---|
|image|image|image|

### README.md updated with instructions:

Preview [here](https://github.com/nimblehq/react-templates/tree/feature/gh-34-add-localization-feature-to-manage-translations/template).

image

### The test uses the mock and passes

image

image

### CODE REVIEW

File: template/src/App.tsx [hunk 1]
```diff
@@ -1,19 +1,22 @@
...
 function App(): JSX.Element {
+  const { t } = useTranslation()
+
   return (
     <div className="app">
       <header className="app-header">
         <img src={logo} className="app-logo" alt="logo" />
         <p>
-          Edit <code>src/App.tsx</code> and save to reload.
+          {t('sample_page.edit')} <code>src/App.tsx</code> {t('sample_page.and_save_to_reload')}.
```
Comment (hoangmirs) [inline template/src/App.tsx:16]: Should we change to use [Interpolation](https://www.i18next.com/translation-function/interpolation) in this line? Or I think we can bring `<code>src/App.tsx</code>` to the translations 🤔

**Review by hoangmirs — COMMENTED** [review]
Rebase required

File: template.json [hunk 2]
```diff
@@ -1,38 +1,42 @@
 {
   "package": {
     "dependencies": {
+      "@nimbl3/eslint-config-nimbl3": "2.1.1",
```
Comment (carryall) [inline template.json:4]: Should we rebase to the latest `develop` branch as this package is the wrong one? [the fix](https://github.com/nimblehq/react-templates/pull/29) was already merged to `develop` branch 😄

File: template/src/App.tsx (hunk 1 above)
Comment (malparty) [reply template/src/App.tsx]: I don't think we should bring HTML code into the translation, but I see the value of providing an example of "interpolation". So what I can suggest is to remove the "<code>" tag and provide an example of interpolation~

Using that translation: `"message": "Edit {{codeSample}} and save to reload",`

It gives: 

| With code tag | Without Code Tag |
|---|---|
|image|image|
|`<p>{t('sample_page.message', { codeSample: '<code>src/App.tsx</code>' })}</p>`|`<p>{t('sample_page.message', { codeSample: '"src/App.tsx"' })}</p>`| 

The `Without code tag` version has been pushed in 1a70ab3 ✅

File: template.json (hunk 2 above)
Comment (malparty) [reply template.json]: Yup! :) Done ;-)

**Review by carryall — APPROVED** [review]

**Review by hoangmirs — APPROVED** [review]

**Pull request merged**